src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -123,4 +125,39 @@ object HttpGoogleServicesDAO {
     storage.objects().get(bucketName, objectKey).executeMediaAsInputStream
   }
 
+  // create a GCS signed url as per https://cloud.google.com/storage/docs/access-control/create-signed-urls-program
+  def getSignedUrl(bucketName: String, objectKey: String) = {
+
+    // generate the string-to-be-signed
+    val verb = "GET"
+    val md5 = ""
+    val contentType = ""
+    val expire = (System.currentTimeMillis() /1000) + 300 // expires 5 minutes (300 seconds) from now
```

Comment:
Nit: spacing. Also, I find it helpful to include the unit for integer durations, so I'd call this `expireSeconds`. Your option.